lib/Runtime/Types/DictionaryTypeHandler.cpp

```diff
@@ -912,7 +912,8 @@ namespace Js
             else if (!(descriptor->Attributes & PropertyConfigurable))
             {
                 // Let/const properties do not have attributes and they cannot be deleted
-                JavascriptError::ThrowCantDelete(propertyOperationFlags, scriptContext, propertyNameString->GetString());
+                JavascriptError::ThrowCantDeleteIfStrictModeOrNonconfigurable(
```

Comment:
>ThrowCantDeleteIfStrictModeOrNonconfigurable [](start = 33, length = 44)

I wonder whether this one (and a few other current callers of ThrowCantDeleteIfStrictModeOrNonconfigurable) should actually call ThrowCantDeleteIfStrictMode because I don't think the flags would be set here to PropertyOperation_ThrowOnDeleteIfNotConfig (but I don't know how to confirm this).

Reply:
Seems safer to call the one that respects both flags. If ThrowOnDeleteIfNotConfig is never set, then it would make no difference in behavior.

---
In reply to: [173517742](https://github.com/Microsoft/ChakraCore/pull/4790#discussion_r173517742) [](ancestors = 173517742)

Reply:
Agree, that it's safer. I was wondering about whether we could limit usage of ThrowCantDeleteIfStrictModeOrNonconfigurable (because I don't like this method :)).

---
In reply to: [173523027](https://github.com/Microsoft/ChakraCore/pull/4790#discussion_r173523027) [](ancestors = 173523027,173517742)